wire/msgtx.go

```diff
@@ -338,6 +432,22 @@ func (msg *MsgTx) BtcEncode(w io.Writer, pver uint32) error {
 		return err
 	}
 
+	// If the pver is set to WitnessVersion, and the Flags field for the
+	// MsgTx aren't 0x00, then this indicates the transaction is to be
+	// encoded using the new witness inclusionary structure defined in
+	// BIP0141.
+	if pver == WitnessVersion {
```

Comment:
Again, this doesn't seem like a correct way to handle it.  I would think it should be looking at `msg.Version >= 2`.